spec-tests/src/test/java/org/candlepin/spec/content/ContentAccessSpecTest.java

```diff
@@ -0,0 +1,861 @@
+        listings = contentListing.elements().next();
+        String updatedX509 = listings.get(0).asText();
+        String updatedContent = listings.get(1).asText();
+        String updatedLastUpdate = contentAccessBody.get("lastUpdate").asText();
+
+        assertEquals(originalX509, updatedX509);
+        assertNotEquals(originalContent, updatedContent);
+        assertNotEquals(originalLastUpdate, updatedLastUpdate);
+    }
+
+    @Test
+    public void shouldOnlyRegenerateContentPartOfContentAccessCertificateWithEnvironmentWithContentChange()
+        throws Exception {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = adminClient.owners().createOwner(Owners.randomSca());
+        String ownerKey = owner.getKey();
+
+        ProductDTO prod = adminClient.ownerProducts().createProductByOwner(ownerKey, Products.random());
+        ContentDTO content = adminClient.ownerContent().createContent(ownerKey, Content.random());
+        adminClient.ownerProducts().addContent(ownerKey, prod.getId(), content.getId(), true);
+        adminClient.owners().createPool(ownerKey, Pools.random(prod));
+
+        EnvironmentDTO env = adminClient.owners().createEnv(ownerKey, Environment.random());
+        promoteContentToEnvironment(adminClient, env.getId(), content, true);
+
+        ConsumerDTO consumer = adminClient.consumers().createConsumer(Consumers.random(owner));
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        JsonNode contentAccessBody = consumerClient.consumers()
+            .getContentAccessBodyJson(consumer.getUuid(), null);
+        JsonNode contentListing = contentAccessBody.get("contentListing");
+        JsonNode listings = contentListing.elements().next();
+        String originalX509 = listings.get(0).asText();
+        String originalContent = listings.get(1).asText();
+        String originalLastUpdate = contentAccessBody.get("lastUpdate").asText();
+
+        content.setName(StringUtil.random("name-"));
+        adminClient.ownerContent().updateContent(ownerKey, content.getId(), content);
+
+        // Sleep a bit to make sure the 'lastUpdate' has more than a second change
+        Thread.sleep(1000);
+
+        contentAccessBody = consumerClient.consumers().getContentAccessBodyJson(consumer.getUuid(), null);
+        contentListing = contentAccessBody.get("contentListing");
+        listings = contentListing.elements().next();
+        String updatedX509 = listings.get(0).asText();
+        String updatedContent = listings.get(1).asText();
+        String updatedLastUpdate = contentAccessBody.get("lastUpdate").asText();
+
+        assertEquals(originalX509, updatedX509);
+        assertNotEquals(originalContent, updatedContent);
+        assertNotEquals(originalLastUpdate, updatedLastUpdate);
+    }
+
+    @Test
+    public void shouldReturnANotModifiedReturnCodeWhenTheDataHasNotBeenUpdatedSinceDate() throws Exception {
```

Comment:
I'm seeing some sporadic results when running this method multiple times. Looking into this, it looks like this may be an issue with the actual logic with the implementation in the ContentAccessManager:

https://github.com/candlepin/candlepin/blob/master/src/main/java/org/candlepin/controller/ContentAccessManager.java#L561-L571